kodilla-patterns/src/main/java/com/kodilla/patterns/proxy/CommandExecutorProxy.java

```diff
@@ -0,0 +1,39 @@
+package com.kodilla.patterns.proxy;
+
+import java.io.IOException;
+
+public class CommandExecutorProxy implements CommandExecutor, User {
+
+    private String name;
+    private String password;
+    private CommandExecutor commandExecutor;
+
+    public CommandExecutorProxy(UserImpl userImpl) {
+        this.name = userImpl.getUserName();
+        this.password = userImpl.getPassword();
+        commandExecutor = new CommandExecutorImpl();
+    }
+
+    @Override
+    public void runCommand(String cmd) throws IOException {
+        //gdy admin ładuj wszystko
+        if (isAdmin()) {
```

Comment:
skoro `name` oraz `password` przekazujesz przez konstruktor i jest niezmienne dla tej instancji klasy `CommandExecutorProxy` to mógłbyś ciało metody `isAdmin()` przenieść do konstruktora i zapisać wynik w zmiennej `boolean isAdmin` i później po prostu jej używać.